src/components/flat-table/flat-table-row/flat-table-row.style.js

```diff
@@ -55,6 +55,14 @@ const StyledFlatTableRow = styled.tr`
   table-layout: fixed;
   width: auto;
 
+  ${({ stickyOffset }) =>
+    stickyOffset > 0 &&
+    css`
+      th {
```

Comment:
I think we need to scope this to only apply within the `FlatTableHeader` somehow as when we use it with `FlatTableRowHeader`s it will mess with the positioning of the columns as they render as a `th`, or maybe use `as` to set them to be `td` when no in the header 🤷‍♂️ 

https://codesandbox.io/s/goofy-noether-j1t09

Reply:
Ah nice one, looks like you've found another existing bug where the sticky styles aren't applied to `FlatTableRowHeader`s within the `FlatTableHead`! 

I'll fix that in this PR as well if it isn't too much extra to add in here